src/ldp/representation/RepresentationMetadata.ts

```diff
@@ -1,47 +1,128 @@
+  }
+
   /**
-   * Optional size of the representation.
+   * Identifier of the resource this metadata is relevant to.
+   * Will update all relevant triples if this value gets changed.
    */
-  byteSize?: number;
+  public get identifier(): NamedNode | BlankNode {
+    return this.id;
+  }
+
+  public set identifier(id: NamedNode | BlankNode) {
+    const quads = this.quads().map((quad): Quad => {
+      if (quad.subject.equals(this.id)) {
+        return createQuad(id, quad.predicate, quad.object, quad.graph);
+      }
+      if (quad.object.equals(this.id)) {
+        return createQuad(quad.subject, quad.predicate, id, quad.graph);
+      }
+      return quad;
+    });
+    this.store = new Store(quads);
+    this.id = id;
+  }
+
   /**
-   * Optional content type of the representation.
+   * @param quads - Quads to add to the metadata.
    */
-  contentType?: string;
+  public addQuads(quads: Quad[]): void {
+    this.store.addQuads(quads);
+  }
+
   /**
-   * Optional encoding of the representation.
+   * @param quads - Quads to remove from the metadata.
    */
-  encoding?: string;
+  public removeQuads(quads: Quad[]): void {
+    this.store.removeQuads(quads);
+  }
+
   /**
-   * Optional language of the representation.
+   * Adds a value linked to the identifier. Strings get converted to literals.
+   * @param predicate - Predicate linking identifier to value.
+   * @param object - Value to add.
    */
-  language?: string;
+  public add(predicate: NamedNode, object: NamedNode | Literal | string): void {
+    this.store.addQuad(this.id, predicate, typeof object === 'string' ? literal(object) : object);
+  }
+
   /**
-   * Optional timestamp of the representation.
+   * Removes the given value from the metadata. Strings get converted to literals.
+   * @param predicate - Predicate linking identifier to value.
+   * @param object - Value to remove.
    */
-  dateTime?: Date;
+  public remove(predicate: NamedNode, object: NamedNode | Literal | string): void {
+    this.store.removeQuad(this.id, predicate, typeof object === 'string' ? literal(object) : object);
+  }
+
   /**
-   * Optional link relationships of the representation.
+   * Removes all values linked through the given predicate.
+   * @param predicate - Predicate to remove.
    */
-  linkRel?: { [id: string]: Set<string> };
+  public removeAll(predicate: NamedNode): void {
+    this.removeQuads(this.store.getQuads(this.id, predicate, null, null));
+  }
+
+  /**
+   * @param predicate - Predicate to get the value for.
+   *
+   * @throws Error
+   * If there are multiple matching values.
+   *
+   * @returns The corresponding value. Undefined if there is no match
+   */
+  public get(predicate: NamedNode): Term | undefined {
+    const quads = this.store.getQuads(this.id, predicate, null, null);
+    if (quads.length === 0) {
+      return;
+    }
+    if (quads.length > 1) {
+      throw new Error(`Multiple results for ${predicate.value}`);
```

Comment:
Or just return first? Depends on contract. Is there another way to get all?

Reply:
I thought there was one but apparently I forgot to add a `getAll` function indeed.